lib/cadet/assessments/assessments.ex

```diff
@@ -744,11 +755,22 @@ defmodule Cadet.Assessments do
         type: question.type
       })
 
-    Repo.insert(
-      answer_changeset,
-      on_conflict: [set: [answer: get_change(answer_changeset, :answer)]],
-      conflict_target: [:submission_id, :question_id]
-    )
+    case Repo.insert(
+           answer_changeset,
+           on_conflict: [set: [answer: get_change(answer_changeset, :answer)]],
+           conflict_target: [:submission_id, :question_id]
+         ) do
+      {:error, struct} ->
+        {:error, struct}
+
+      {:ok, struct} ->
+        # ChatKit - create chatrooms if they don't exist
+        create_rooms(submission)
```

Comment:
Why are you creating rooms here. This function is called when students save an answer on the frontend

Reply:
As we are now depending on an external service to host tutor comments, rooms are currently being created at three points - when answers are created, when students edit their answers and when cadets access answer in the grading view. In the event Chatkit crashes, the last two points serve as failsafes to ensure comments get to the students.

Reply:
An alternative to the current failsafes, if you would prefer, is to write a script to create all chatrooms after Chatkit restores its services, if it crashes. However, do note that the script needs to be run manually, and each request to their server averages 400-500ms.

Reply:
> An alternative to the current failsafes, if you would prefer, is to write a script to create all chatrooms after Chatkit restores its services, if it crashes. The script will be scheduled to run daily to check for answers with an empty comment field (no chatroom), and make the request to create a new room.

Yes, i think this will be better.

Reply:
Changed with most recent [commit](https://github.com/source-academy/cadet/pull/406/commits/2b01b523689b3609b1621d56a9f4f291b29f7758). Script is scheduled to run at 1am, when hopefully there will be fewer students active online.